internal/meta/config_info.go

```diff
@@ -170,10 +171,14 @@ func (cfgs ConfigInfos) addReferenceDependency() error {
 
 			var dependingResourceIdsWithoutSelf []string
 			for _, id := range dependingResourceIds[:] {
-				if id == cfg.AzureResourceID.String() {
+				if id.String() == cfg.AzureResourceID.String() {
 					continue
 				}
-				dependingResourceIdsWithoutSelf = append(dependingResourceIdsWithoutSelf, id)
+				// if cfg is parent of `id` resource, we should skip, or it will cause circular dependency, so skip parent depends on sub resources
+				if cfg.AzureResourceID.Equal(id.Parent()) {
+					continue
+				}
+				dependingResourceIdsWithoutSelf = append(dependingResourceIdsWithoutSelf, id.String())
```

Comment:
The cyclic dependency is introduced in cases like importing a NSG with multiple NSRs, there will be NSR ids in the NSG body, whilst NSG is the parent of NSR, in which case the NSR ids should be regarded as reference from NSG.

Above isn't the case when we up hold the assumption that the `cfg` only contains those `Required`/`Optional` properties, but ignore the `Comptued`-only attributes. In above case, the NSR ids are comptued only attributes, which shouldn't be visited.

Reply:
Discussed offline, the API didn't specified, e.g., the NSR id as read-only. This is the best off effort we can do to avoid that poor API definition.